### ISSUE TITLE
File change throws "Cannot read property file of null" on NavigationAndHistory/main.js

### ISSUE THREAD

**Issue by petetnt** [adobe/brackets#12792]
### Prerequisites
- [x] Can you reproduce the problem with `Debug \ Reload Without Extensions`?
- [x] Did you perform a cursory search to see if your bug or enhancement is already reported?
- [x] Did you read the [Troubleshooting guide](https://github.com/adobe/brackets/wiki/Troubleshooting)?
### Description

For some time now (~5 days?) I have been getting the following errors every time I switch files. The fix would be trivial (checking for `file` first in the if-clause) but not sure yet what broke it in the first place. Need to do some debugging.
### Steps to Reproduce
1. Open a file

**Expected behavior:** [What you expected to happen]
File opens

**Actual behavior:** [What actually happened]

File is changed, but the following erros are thrown:

``` js
 Exception in 'currentFileChange' listener on Object TypeError: Cannot read property 'file' of null TypeError: Cannot read property 'file' of null
    at file:///Applications/Brackets.app/Contents/dev/src/extensions/default/NavigationAndHistory/main.js:589:27
    at t (file:///Applications/Brackets.app/Contents/dev/src/thirdparty/lodash.js:6:249)
    at Function.<anonymous> (file:///Applications/Brackets.app/Contents/dev/src/thirdparty/lodash.js:28:324)
    at _addToMROFList (file:///Applications/Brackets.app/Contents/dev/src/extensions/default/NavigationAndHistory/main.js:588:23)
    at handleCurrentFileChange (file:///Applications/Brackets.app/Contents/dev/src/extensions/default/NavigationAndHistory/main.js:772:10)
    at Object.trigger (file:///Applications/Brackets.app/Contents/dev/src/utils/EventDispatcher.js:229:40)
    at file:///Applications/Brackets.app/Contents/dev/src/view/MainViewManager.js:1106:29
    at Pane.trigger (file:///Applications/Brackets.app/Contents/dev/src/utils/EventDispatcher.js:229:40)
    at Pane._notifyCurrentViewChange (file:///Applications/Brackets.app/Contents/dev/src/view/Pane.js:890:14)
    at Pane.showView (file:///Applications/Brackets.app/Contents/dev/src/view/Pane.js:1202:14)
    at _showEditor (file:///Applications/Brackets.app/Contents/dev/src/editor/EditorManager.js:567:14)
    at Object.openDocument (file:///Applications/Brackets.app/Contents/dev/src/editor/EditorManager.js:639:13)
    at _edit (file:///Applications/Brackets.app/Contents/dev/src/view/MainViewManager.js:1198:23)
    at Object.<anonymous> (file:///Applications/Brackets.app/Contents/dev/src/view/MainViewManager.js:1262:21)
    at j (file:///Applications/Brackets.app/Contents/dev/src/thirdparty/jquery-2.1.3.min.js:2:26911)
    at Object.k.fireWith [as resolveWith] (file:///Applications/Brackets.app/Contents/dev/src/thirdparty/jquery-2.1.3.min.js:2:27724)
utils/EventDispatcher.js:232 

Assertion failed:utils/EventDispatcher.js:232 trigger
```
### Versions

OSX El Capitan 10.11.6
Commit: a340c30, 1.8 pre-release shell but also occurs on 1.7

**Comment by petetnt** [comment]
Suddenly it stopped doing that, without restarting/reloading or restarting Brackets 🤔 🤔

**Comment by ficristo** [comment]
Related https://github.com/adobe/brackets/issues/12629

**Comment by petetnt** [comment]
@ficristo Hah, I like how the chain goes on like my issue - your issue - my issue... 😆

**Comment by corfeur12** [comment]
I have been having this exact same problem too, but it has not stopped for me. I am using Windows 10 Version 1511 (OS Build 10586.589) on a 64bit processor.

**Comment by ficristo** [comment]
@corfeur12 have you tryed `File \ Open Recent Files...`  and then press the `Clear` button?
I don't think it is a real fix, but as a temporary workaround could help.

**Comment by corfeur12** [comment]
@ficristo I have just done that and it seems to have fixed it, thank you

**Comment by swmitra** [comment]
@zaggino @petetnt @ficristo I am pushing this issue to 1.9 milestone.

**Comment by swmitra** [comment]
This will be fixed as part of #13418 .

**Comment by petetnt** [comment]
This line (https://github.com/adobe/brackets/pull/13418/files#diff-649cccfc4b4582653f0bc2c44978e56cR650) right?

**Comment by swmitra** [comment]
Yes @petetnt

**Comment by swmitra** [comment]
@petetnt #13418 is merged and available as part of pre-release 3 builds. Can you please verify this defect and close if not reproducible?

**Comment by petetnt** [comment]
Sure thing @swmitra, I haven't figured out repro steps (!) for this one, but I haven't had the issue anymore when I fixed it locally. Closing as fixed, nice work.

**Comment by petetnt** [comment]
Still not fixed after all. Something is still inserting null values to `mroflist` and this is throwing again:

``` js
        // Check existing list for this doc path and pane entry
        var index = _.findIndex(_mrofList, function (record) {
            return (record.file === filePath && record.paneId === paneId);
        });
```

Switching branches on `git` while brackets is open is a good way to replicate this.

**Comment by swmitra** [comment]
Should be fixed now by #13495 :face_with_head_bandage:
